src/app/dynamic-forms/cinchy-dynamic-forms.component.ts

```diff
@@ -432,105 +469,163 @@ export class CinchyDynamicFormsComponent implements OnInit, OnChanges, OnDestroy
-      const cinchyColumn: ICinchyColumn = new CinchyColumn(formFieldMetadataQueryResult[i].ColumnId,
-        formFieldMetadataQueryResult[i].TableId, formFieldMetadataQueryResult[i].Table,
-        formFieldMetadataQueryResult[i].Domain,
-        formFieldMetadataQueryResult[i].ColumnName, formFieldMetadataQueryResult[i].ColumnType,
-        isMandatory, formFieldMetadataQueryResult[i].ColumnMaxLength,
-        formFieldMetadataQueryResult[i].LinkTargetColumnId, formFieldMetadataQueryResult[i].LinkTargetColumnNName, allowMultiple,
-        validationExpression, minValue, canEdit, canView, formFieldMetadataQueryResult[i].LinkTargetTableId,
-        col, formFieldMetadataQueryResult[i].choiceOptions, formFieldMetadataQueryResult[i].FormFieldsJsonData, formFieldMetadataQueryResult[i].dataFormatType, false,
-        viewOnly, linkFieldId, formFieldMetadataQueryResult[i].IsDisplayColumn, attachedFileName, formFieldMetadataQueryResult[i].FileNameColumn,
-        formFieldMetadataQueryResult[i].dropdownFilter, formFieldMetadataQueryResult[i].totalTextAreaRows, formFieldMetadataQueryResult[i].numberFormatter,
-        formFieldMetadataQueryResult[i].attachmentURL, formFieldMetadataQueryResult[i].uploadURL, childFormParentId,
-        childFormLinkId, formFieldMetadataQueryResult[i].doNotWrap
+
+      const isMandatory = formFieldMetadata.overrideMandatory ? false : formFieldMetadata.ColumnIsMandatory;
+      const cinchyColumn: ICinchyColumn = new CinchyColumn(formFieldMetadata.ColumnId,
+        formFieldMetadata.TableId, formFieldMetadata.Table,
+        formFieldMetadata.Domain,
+        formFieldMetadata.ColumnName, formFieldMetadata.ColumnType,
+        isMandatory, formFieldMetadata.ColumnMaxLength,
+        formFieldMetadata.LinkTargetColumnId, formFieldMetadata.LinkTargetColumnNName, allowMultiple,
+        validationExpression, minValue, canEdit, canView, formFieldMetadata.LinkTargetTableId,
+        col, formFieldMetadata.choiceOptions, formFieldMetadata.FormFieldsJsonData, formFieldMetadata.dataFormatType, false,
+        viewOnly, linkFieldId, formFieldMetadata.IsDisplayColumn, attachedFileName, formFieldMetadata.FileNameColumn,
+        formFieldMetadata.dropdownFilter, formFieldMetadata.totalTextAreaRows, formFieldMetadata.numberFormatter,
+        formFieldMetadata.attachmentURL, formFieldMetadata.uploadURL, childFormParentId,
+        childFormLinkId, formFieldMetadata.doNotWrap, displayFormat
       );
       const dropdownDataset: DropdownDataset = null;
+      const childFormId: number = formFieldMetadata.ChildFormId;
+      if(displayFormat && cinchyColumn.dataType==='Date and Time'){
+        let existingFormat = sessionStorage.getItem('displayFormat');
+        if(existingFormat !== displayFormat){
+          window.location.reload();
+        }
+      }
+      displayFormat && sessionStorage.setItem('displayFormat', displayFormat);
```

Comment:
Can you explain to me the purpose of using sessionStorage for display format?

Reply:
The configuration for the dateformat in the material datepicker was preloaded and inorder to make that dynamic and fetch it from the api, we are using session storage

Reply:
removed sessionStorage logic